sys-libs/glibc/glibc-2.33-r8.ebuild

```diff
@@ -1392,7 +1392,7 @@ glibc_do_src_install() {
 		sed -i "${nscd_args[@]}" "${ED}"/etc/init.d/nscd
 
 		use systemd && systemd_dounit nscd/nscd.service
-		systemd_newtmpfilesd nscd/nscd.tmpfiles nscd.conf
+		newtmpfiles nscd/nscd.tmpfiles nscd.conf
```

Comment:
Confused.
Gentoo uses the default `newtmpfiles`.
But Flatcar has been applying its own [patch](https://github.com/flatcar-linux/coreos-overlay/pull/1087/commits/8d040f93c289cd696db714ac8ae92d55e04171ac) to replace `newtmpfiles` with `systemd_newtmpfilesd`.
Does it mean that we do not need to keep the Flatcar patch for glibc?

Reply:
So it seems that the reason for replacing `newtmpfiles` with `systemd_newtmpfilesd` was to avoid circular dep between glibc and system, because `tmpfiles` eclass was adding `virtual/tmpfiles` into dependencies (`virtual/tmpfiles` pulled in systemd), while `systemd` eclass was not adding any new deps by inheriting it. The `tmpfiles` eclass got recently a new option (`TMPFILES_OPTIONAL`) that allows us to skip adding `virtual/tmpfiles`, so we can drop this change from glibc.

Reply:
Ok, I believe you are correct. We can drop the newtmpfiles from the glibc patch.
Still, I am afraid, the next person who upgrades glibc could probably get into a rabbit hole to figure out which parts of the big chunk of the [patch](https://github.com/flatcar-linux/coreos-overlay/pull/1087/commits/8d040f93c289cd696db714ac8ae92d55e04171ac) are still valid.
Ideally we should have 2 commits for glibc, one for plain sync commit with Gentoo, the other for all Flatcar changes.

Same applies maybe for other ones in the other 20 commits.

Also I am not sure I like bumping the revision like `-r7` to `-r8` for every commit, especially for glibc.
In the past revisions in glibc 2.33 in Gentoo had quite special meaning. `-r5` was to address specific CVEs, and `-r7` was to address other CVEs, etc.
Worse, that revision number did not match with `PATCH_VER` to be used for the patch tarball.
Now we bumped the revision to -r8, and let's say Gentoo suddenly releases 2.33-r8 tomorrow. That must be fun.
Not sure that would be the case as well for glibc 2.34 though.

Reply:
> Ok, I believe you are correct. We can drop the newtmpfiles from the glibc patch. Still, I am afraid, the next person who upgrades glibc could probably get into a rabbit hole to figure out which parts of the big chunk of the [patch](https://github.com/flatcar-linux/coreos-overlay/pull/1087/commits/8d040f93c289cd696db714ac8ae92d55e04171ac) are still valid. Ideally we should have 2 commits for glibc, one for plain sync commit with Gentoo, the other for all Flatcar changes.
> 
> Same applies maybe for other ones in the other 20 commits.

Sure, I can do that.

> 
> Also I am not sure I like bumping the revision like `-r7` to `-r8` for every commit, especially for glibc. In the past revisions in glibc 2.33 in Gentoo had quite special meaning. `-r5` was to address specific CVEs, and `-r7` was to address other CVEs, etc. Worse, that revision number did not match with `PATCH_VER` to be used for the patch tarball. Now we bumped the revision to -r8, and let's say Gentoo suddenly releases 2.33-r8 tomorrow. That must be fun. Not sure that would be the case as well for glibc 2.34 though.

I would say that when an ebuild is modified, its revision should be bumped. But it's icky in case of forked packages like glibc, true. Maybe we could use one of two ways out of it:

1. For packages that are forked from gentoo, multiply their revision by 10 or 100 and bump the revision every time we modify it. Cons: things like `PATCH_VER` or generally the `PVR` and `PF` variables don't have the contents expected by the build (for example it expected `PVR` to be `2.33-r7`, not `2.33-r701`).
2. Modify portage by adding subrevision? Something like `glibc-2.33-r7.1`. That would still be newer than `r7`, but older than `r8`. Cons: another patch to carry in portage, I suppose.

My proposal would be that for now let's not worry about it and then get the revbumping sorted out in new year.

Reply:
> For packages that are forked from gentoo, multiply their revision by 10 or 100 and bump the revision every time we modify it. Cons: things like PATCH_VER or generally the PVR and PF variables don't have the contents expected by the build (for example it expected PVR to be 2.33-r7, not 2.33-r701).
> Modify portage by adding subrevision? Something like glibc-2.33-r7.1. That would still be newer than r7, but older than r8. Cons: another patch to carry in portage, I suppose.

TBH both look ugly. ;-)

Ideally we should update each ebuild to the next version (like glibc 2.34) as often as possible, without having to worry about damn complicated revision numbers.
And we are already doing that for many packages.
However, this is glibc, one of the heart of OS. That's why frequent version bumps are not trivial.
I would also hope the Gentoo glibc maintainers stop using different PATCH_VER in the first place. But anyway there is nothing we can do about that.

Reply:
> > For packages that are forked from gentoo, multiply their revision by 10 or 100 and bump the revision every time we modify it. Cons: things like PATCH_VER or generally the PVR and PF variables don't have the contents expected by the build (for example it expected PVR to be 2.33-r7, not 2.33-r701).
> > Modify portage by adding subrevision? Something like glibc-2.33-r7.1. That would still be newer than r7, but older than r8. Cons: another patch to carry in portage, I suppose.
> 
> TBH both look ugly. ;-)

:grin:

I had a look at [ebuild revisions doc](https://devmanual.gentoo.org/general-concepts/ebuild-revisions/index.html) and actually not all changes warrant a revbump. TIL. I also had an idea of using [the `_p` suffix in package version](https://devmanual.gentoo.org/ebuild-writing/file-format/index.html#file-naming-rules), but that's worse than the above, so I'll shut up now. :)

> 
> Ideally we should update each ebuild to the next version (like glibc 2.34) as often as possible, without having to worry about damn complicated revision numbers. And we are already doing that for many packages. However, this is glibc, one of the heart packages of Linux OS. That's why frequent version bumps are not trivial. I would also hope the Gentoo glibc maintainers stop using different PATCH_VER in the first place. But anyway there is nothing we can do about that.

Alright, I'll drop the revbump for glibc package then. The change I made is rather trivial, anyway.